File: chris_cunningham/day01.rs
use std::fs;
use itertools::Itertools;

fn get_answer(d: &Vec<i32>, r: usize) -> i32 {
    d.iter().cloned().combinations(r)
        .find(|x| x.iter().sum::<i32>() == 2020)
        .unwrap()
        .iter()
        .product()
}

pub fn solve() {
    let data: Vec<i32> = fs::read_to_string("inputs/day01.txt")
        .unwrap()
        .lines()
        .filter_map(|s| s.parse::<i32>().ok())
        .collect();

    println!("part a: {}", get_answer(&data, 2));
    println!("part b: {}", get_answer(&data, 3));
}

File: chris_cunningham/day01.py
from typing import List
from itertools import combinations
from math import prod


def get_answer(d: List[int], r: int):
    return next(prod(d) for d in combinations(d, r) if sum(d) == 2020)


with open("inputs/day01.txt", 'r') as f:
    data = [int(i) for i in f.readlines()]
    print(f"part a: {get_answer(data, 2)}")
    print(f"part b: {get_answer(data, 3)}")
